server/channels/app/plugin_hooks_test.go

```diff
@@ -1985,11 +1985,21 @@ func TestUserHasJoinedChannel(t *testing.T) {
 		require.Nil(t, appErr)
 
 		assert.EventuallyWithT(t, func(t *assert.CollectT) {
-			posts, appErr := th.App.GetPosts(channel.Id, 0, 1)
+			posts, appErr := th.App.GetPosts(channel.Id, 0, 30)
 
 			require.Nil(t, appErr)
 			assert.True(t, len(posts.Order) > 0)
-			assert.Equal(t, fmt.Sprintf("Test: User %s joined %s", user2.Id, channel.Id), posts.Posts[posts.Order[0]].Message)
+
+			found := false
+			for _, post := range posts.Posts {
```

Comment:
@hmhealey it was still flaky because it always tried to check in the vary last post. This won't work with parallel tests, for example. So to fix this once and for all, I'm fetching a lot more posts and trying to find the desired post in them and it's working fine for now.